sklearn/metrics/classification.py

```diff
@@ -596,72 +597,135 @@ def jaccard_similarity_score(y_true, y_pred, normalize=True,
+        determines the type of averaging performed on the data:
+
+        ``'binary'``:
+            Only report results for the class specified by ``pos_label``.
+            This is applicable only if targets (``y_{true,pred}``) are binary.
+        ``'micro'``:
+            Calculate metrics globally by counting the total true positives,
+            false negatives and false positives.
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted
+            mean.  This does not take label imbalance into account.
+        ``'weighted'``:
+            Calculate metrics for each label, and find their average, weighted
+            by support (the number of true instances for each label). This
+            alters 'macro' to account for label imbalance.
+        ``'samples'``:
+            Calculate metrics for each instance, and find their average (only
+            meaningful for multilabel classification).
+
     normalize : bool, optional (default=True)
         If ``False``, return the sum of the Jaccard similarity coefficient
         over the sample set. Otherwise, return the average of Jaccard
-        similarity coefficient.
+        similarity coefficient. ``normalize`` is only applicable when
+        ``average='samples'``.
 
     sample_weight : array-like of shape = [n_samples], optional
         Sample weights.
 
     Returns
     -------
-    score : float
-        If ``normalize == True``, return the average Jaccard similarity
-        coefficient, else it returns the sum of the Jaccard similarity
-        coefficient over the sample set.
-
-        The best performance is 1 with ``normalize == True`` and the number
-        of samples with ``normalize == False``.
+    score : float (if average is not None) or array of floats, shape =\
+            [n_unique_labels]
 
     See also
     --------
     accuracy_score, hamming_loss, zero_one_loss
 
     Notes
     -----
-    In binary and multiclass classification, this function is equivalent
-    to the ``accuracy_score``. It differs in the multilabel classification
-    problem.
+    :func:`jaccard_similarity_score` may be a poor metric if there are no
+    positives for some samples or classes.
 
     References
     ----------
     .. [1] `Wikipedia entry for the Jaccard index
            <https://en.wikipedia.org/wiki/Jaccard_index>`_
 
-
     Examples
     --------
+    >>> import numpy as np
     >>> from sklearn.metrics import jaccard_similarity_score
-    >>> y_pred = [0, 2, 1, 3]
-    >>> y_true = [0, 1, 2, 3]
-    >>> jaccard_similarity_score(y_true, y_pred)
+
+    In the multilabel case:
+
+    >>> y_true = np.array([[1, 0, 1], [0, 0, 1], [1, 1, 1]])
+    >>> y_pred = np.array([[0, 1, 1], [1, 1, 1], [0, 0, 1]])
+    >>> jaccard_similarity_score(y_true, y_pred, average='samples')
+    ... # doctest: +ELLIPSIS
+    0.33...
+    >>> jaccard_similarity_score(y_true, y_pred, average='micro')
+    ... # doctest: +ELLIPSIS
```

Comment:
I think this is redundant, it's already set above (and it generates the odd empty `... ` line in the output).

Reply:
I think these flags are per-statement, so I don't see how "it's already set above"

Reply:
The scope of those flags are at least per-block, example:

https://github.com/scikit-learn/scikit-learn/blob/8d10ba0fe912dee1d67d0d3c2d3d3b6765047d16/sklearn/covariance/empirical_covariance_.py#L128-L132